### ISSUE TITLE
Send report to a github page report

### ISSUE THREAD

**Issue by courentin** [elementary-data/elementary#373]
**Is your feature request related to a problem? Please describe.**

Sending reports through S3 is great but by default the s3 website will be publicly accessible. They are way to make it private but they seems pretty complex for a single page website.

**Describe the solution you'd like**

We could host the report on a github page repository, that would ease access control.
The idea would be to push the html file to a github repository.

**Comment by elongl** [comment]
Hi @courentin! Thanks for the great suggestion.
I'd like to share a couple of things if that's okay.

First off, you can also send your report to Slack or to a GCS bucket if those fit you better.
Regarding the S3, I agree that [setting up a website](https://docs.aws.amazon.com/AmazonS3/latest/userguide/WebsiteHosting.html) could be tedious,
however, you can also upload the report to the bucket with a fixed name like `report.html` and grant access to that file.

Regarding your request, it seems like it should already be possible to do that.
Basically what you'd need to do is create a GitHub workflow in your GitHub Pages repository that does the following steps:
1.  Run [Elementary's action](https://github.com/marketplace/actions/run-elementary).
2. Run the [add-and-commit action](https://github.com/marketplace/actions/add-commit) to commit the report to the repository.

Afterwards, you should see the report being updated in your website.

How does that sound to you?
Care to give it a shot and share some insights?

Thanks a lot!
Feel free to let me know if you have any questions.

**Comment by courentin** [comment]
Hello @elongl thanks for your answer!

Yes I can definitely do use an action for this, that would work perfectly (and it's actually what's I'm doing for the dbt doc).
That was more an open question about "would there be a benefit of including it directly in the send-report command".

I have not strong opinions on it as above's solution works perfectly fine for me!

**Comment by Maayan-s** [comment]
@elongl Maybe we should add this use case to our docs?

**Comment by elongl** [comment]
@Maayan-s Sounds like a great idea!

@courentin Would you perhaps be interested in adding a section to Elementary's docs regarding how you integrated Elementary's report with your GitHub Pages? I think it can be an interesting use-case for a lot of users and it seems like you really nailed this mechanism. Let me know what you think 😄 

You can do it by clicking the _Edit this page_ on the [Share observability report](https://docs.elementary-data.com/quickstart/share-report-ui) page.

**Comment by elongl** [comment]
I recently noted that there's a very popular [action](https://github.com/marketplace/actions/deploy-to-github-pages) that deploys to GitHub Pages. Maybe worth looking into it and understand the difference.
